src/commands/ext-dev-deprecate.ts

```diff
@@ -4,77 +4,141 @@ import * as semver from "semver";
+  versionPredicate: string,
+  options: ExtDevDeprecateOptions
+) {
+  const { publisherId, extensionId, version } = extensionRef;
+  if (version) {
+    throw new FirebaseError(
+      `The input extension reference must be of the format ${clc.bold(
+        "<publisherId>/<extensionId>"
+      )}. Version should be supplied in the version predicate argument.`
+    );
+  }
+  if (!publisherId || !extensionId) {
+    throw new FirebaseError(
+      `Error parsing publisher ID and extension ID from extension reference '${clc.bold(
+        refs.toExtensionRef(extensionRef)
+      )}'. Please use the format '${clc.bold("<publisherId>/<extensionId>")}'.`
     );
-    utils.logLabeledSuccess(logPrefix, "successfully deprecated extension version(s).");
-  });
+  }
+
+  // Error out if --delete and version predicate isn't *
+  const { comparator, targetSemVer } = parseVersionPredicate(versionPredicate);
+  const filter = `id${comparator}"${targetSemVer}"`;
+  const extensionVersions = await listExtensionVersions(refs.toExtensionRef(extensionRef), filter);
+  const filteredExtensionVersions = extensionVersions
+    .sort((ev1, ev2) => {
+      return -semver.compare(ev1.spec.version, ev2.spec.version);
+    })
+    .filter((extensionVersion) => {
+      if (extensionVersion.state === "DEPRECATED" && !options.force) {
+        return false;
+      }
+      const message =
+        extensionVersion.state === "DEPRECATED" ? ", will overwrite deprecation message" : "";
+      utils.logLabeledBullet(extensionVersion.ref, extensionVersion.state + message);
+      return true;
+    });
+  if (filteredExtensionVersions.length > 0) {
+    const consent = await confirm({
+      default: false,
+      force: options.force,
+      nonInteractive: options.nonInteractive,
+    });
+    if (!consent) {
+      throw new FirebaseError("Deprecation canceled.");
+    }
+  } else {
+    throw new FirebaseError("No extension versions matched the version predicate.");
+  }
+  await utils.allSettled(
+    filteredExtensionVersions.map(async (extensionVersion) => {
+      await deprecateExtensionVersion(extensionVersion.ref, options.message);
+    })
+  );
+  utils.logLabeledSuccess(logPrefix, "successfully deprecated extension version(s).");
+}
+
+async function deleteExt(
+  extensionRef: refs.Ref,
+  versionPredicate: string,
+  options: ExtDevDeprecateOptions
+) {
+  if (versionPredicate) {
+    throw new FirebaseError("");
+  }
+  const extRef = refs.toExtensionRef(extensionRef);
+  utils.logLabeledWarning(
+    logPrefix,
+    "If you delete this extension, developers won't be able to install it. " +
+      "For developers who currently have this extension installed, " +
+      "it will continue to run and will appear as unpublished when " +
+      "listed in the Firebase console or Firebase CLI."
+  );
+  utils.logLabeledWarning(
+    logPrefix,
+    "This is a permanent action" +
+      `Once deleted, you may never use the extension name '${clc.bold(extRef)}' again.`
```

Comment:
Nit: Can you refactor this string as a single literal? I think this might currently read "permanent actionOnce deleted".